api/pkg/apis/v1alpha1/utils/utils_test.go

```diff
@@ -404,3 +404,85 @@ func TestToInterfaceMap(t *testing.T) {
 	assert.Equal(t, "bar", m2["foo"])
 	assert.Equal(t, "123", m2["abc"])
 }
+
+func TestReadEmptyStringProperty(t *testing.T) {
+	m := map[string]interface{}{
+		"a": map[string]interface{}{
+			"b": map[string]interface{}{
+				"c": "foo",
+			},
+		},
+	}
+	m2, ok := JsonParseProperty(m, "")
+	assert.False(t, ok)
+	assert.Equal(t, nil, m2)
+}
+
+func TestReadRootProperty(t *testing.T) {
+	m := map[string]interface{}{
+		"a": map[string]interface{}{
+			"b": map[string]interface{}{
+				"c": "foo",
+			},
+		},
+	}
+	m2, ok := JsonParseProperty(m, ".")
+	assert.True(t, ok)
+	assert.Equal(t, m, m2)
+}
+
+func TestReadNestedJsonStringProperty(t *testing.T) {
+	value := "123"
+	m := map[string]interface{}{
+		"a": map[string]interface{}{
+			"b": map[string]interface{}{
+				"c": value,
+			},
+		},
+	}
+	m2, ok := JsonParseProperty(m, "a.b.c")
+	assert.True(t, ok)
+	assert.Equal(t, value, m2)
+}
```

Comment:
I'm curious if there's any field can contain "." like
```go
m := map[string]interface{}{
		"a": map[string]interface{}{
			"b.c": map[string]interface{}{
				"c": value,
			},
		},
	}
```

Reply:
Updated to allow the dot name scenario